code/game/machinery/OpTable.dm

```diff
@@ -148,6 +181,20 @@
 
 	add_fingerprint(user)
 
+	update()
+
+	if(!pulsing)
+		while(victim.loc == src.loc && victim.lying && victim.pulse != PULSE_NONE && victim.stat != DEAD && !victim.timestopped)
+			pulsing = TRUE
+			var/pulsespeed = victim.get_pulsespeed()
+			if(pulsespeed)
+				if(computer)
+					playsound(computer.loc, 'sound/machines/Heartbeat.ogg', 50)
+					computer.icon_state = "operating-living"
+				icon_state = "table2-active"
+				sleep(max(1,pulsespeed))
```

Comment:
Do not do this. The machine subsystem expects process() to return expediently -- it doesn't spawn() the call or anything. The sleep here fucks everything up.

I think you should instead do a separate infinite loop that's spawned when someone is put on the table and terminates when they're taken off of it (or die, or some other sensible conditions).

Reply:
isn't this under take_victim() though, with no relation to process()?

Reply:
oh, you're right. I got tricked by Github collapsing the diff below process() omitting the actual proc definition this was under.

Reply:
That being said I think you should still `spawn()` this loop so that the proc will return normally instead of having to wait for the loop to finish.